addons/survey/views/survey_templates_management.xml

```diff
@@ -118,12 +118,29 @@
                             </div>
                             <div class="row">
                                 <div class="col-12 col-md-4 offset-md-4 text-center">
-                                    <input id="session_code" type="text" placeholder="e.g. 4812" autocomplete="off"
+                                    <input t-if="session_code" id="session_code" type="text" placeholder="e.g. 4812" autocomplete="off" class="form-control o_survey_question_text_box font-weight-bold bg-transparent text-primary text-center rounded-0 p-2 w-100" t-att-value="session_code"/>
+                                    <input t-else="" id="session_code" type="text" placeholder="e.g. 4812" autocomplete="off"
                                            class="form-control o_survey_question_text_box font-weight-bold bg-transparent text-primary text-center rounded-0 p-2 w-100"/>
                                 </div>
                                 <div class="col-12 col-md-4 offset-md-4 text-center o_survey_error text-danger pt-2" role="alert">
-                                    <span class="o_survey_session_error_invalid_code d-none">Code is incorrect.</span>
-                                    <span class="o_survey_session_error_closed d-none">Session is finished.</span>
+                                    <span t-if="error == 'survey_wrong'" class="o_survey_session_error_invalid_code">Code is incorrect.</span>
+                                    <span t-else="" class="o_survey_session_error_invalid_code d-none">Code is incorrect.</span>
+                                    <span t-if="error == 'survey_session_closed'" class="o_survey_session_error_closed">Session is finished.</span>
+                                    <span t-else="" class="o_survey_session_error_closed d-none">Session is finished.</span>
+                                    <span t-if="error == 'survey_session_not_launched'" class="o_survey_session_error_not_launched">The session did not start yet.</span>
+                                    <span t-else="" class="o_survey_session_error_not_launched d-none">
+                                        The session did not start yet.
+                                    </span>
+                                    <span t-if="user_can_launch" t-att-survey_id="survey_id" class="o_survey_session_launch_session btn btn-link p-0">
```

Comment:
You should be able to test the group directly in the template.
Using a "groups" attribute.